src/controller/user/login.controller.ts

```diff
@@ -0,0 +1,57 @@
+import { envVar } from "@/index";
+import { LoginUserProps } from "@/models/login/type";
+import LoginService from "@/service/login.service";
+import UserService from "@/service/user.service";
+import {
+  getInternalServerErrorResponse,
+  getOKResponse,
+  getUnauthorizedResponse,
+} from "@/utils/helpers/response";
+import { IReq, IRes } from "@/utils/interfaces/express.interface";
+import jwt from "jsonwebtoken";
+
+class LoginController {
+  private loginService: LoginService;
+  constructor() {
+    this.loginService = new LoginService();
+  }
+
+  loginUser = async (req: IReq, res: IRes) => {
+    const inputData = { ...req.body } as LoginUserProps;
+    try {
+      const loginData = await this.loginService.findUserByCredentials(
+        inputData.userNameOrEmail,
+        inputData.password
+      );
+
+      if (!!loginData.length) {
+        const { userName, email, _id } = loginData[0];
+        const accessToken = jwt.sign(
+          {
+            email: inputData.userNameOrEmail,
+          },
+          envVar.JSON_SECRET_KEY,
+          { expiresIn: "1h" }
+        );
+        const loginObj = {
```

Comment:
add a type to this `loginObj`. Probably what `userLogin` service function has in its input Type.